lib/klaviyo/client.rb

```diff
@@ -96,10 +101,18 @@ def self.check_private_api_key_exists()
       end
     end
 
-    def self.check_public_api_key_exists()
+    def self.check_public_api_key_is_valid()
       if !Klaviyo.public_api_key
         raise KlaviyoError.new(NO_PUBLIC_API_KEY_ERROR)
       end
+      is_not_private_key = Klaviyo.public_api_key =~ /pk_\w{34}$/
+      valid_public_key = Klaviyo.public_api_key =~ /\w{6}$/
+      if is_not_private_key == 0
```

Comment:
Do you think it's valuable to not store the comparison in memory?
We use the variable only once and I don't think this affects readability a lot either since the expression isn't huge.
```suggestion
      if ( Klaviyo.public_api_key =~ /pk_\w{34}$/ ) == 0
```